test/unit/transformRemappings.spec.ts

```diff
@@ -5,35 +5,176 @@ import mock from 'mock-fs';
+                                `;
+
+      const transformedContent = transformRemappings(fileContent);
 
-  it('should transform imports with remappings', () => {
+      expect(transformedContent).to.include(`import '../../../../../interfaces/ITest.sol';`);
+      expect(transformedContent).to.include(`import {ITest} from '../../../../../interfaces/ITest.sol';`);
+      expect(transformedContent).to.include(`import {ITest, ITest2} from '../../../../../interfaces/ITest.sol';`);
+      expect(transformedContent).to.include(`import {IBar as IFoo, ITest} from '../../../../../interfaces/ITest.sol';`);
+      expect(transformedContent).to.include(`import * as Interfaces '../../../../../interfaces/ITest.sol';`);
+      expect(transformedContent).to.include(`import "../../../../../interfaces/ITest.sol";`);
+      expect(transformedContent).to.include(`import {ITest} from "../../../../../interfaces/ITest.sol";`);
+      expect(transformedContent).to.include(`import {ITest, ITest2} from "../../../../../interfaces/ITest.sol";`);
+      expect(transformedContent).to.include(`import {IBar as IFoo, ITest} from "../../../../../interfaces/ITest.sol";`);
+      expect(transformedContent).to.include(`import * as Interfaces "../../../../../interfaces/ITest.sol";`);
+    });
+
+    it('should remove node_modules from import paths', function () {
+      const fileContent = `
+      import '../../../node_modules/some-package/Contract.sol';
+      import 'node_modules/another-package/Token.sol';
+      import {FixedPointMathLib} from 'node_modules/solmate/src/utils/FixedPointMathLib.sol';
+      import {FixedPointMathLib} from "node_modules/solmate/src/utils/FixedPointMathLib.sol";
+      import {FixedPointMathLib as FPL} from "node_modules/solmate/src/utils/FixedPointMathLib.sol";
+      import * as FPL from "../node_modules/solmate/src/utils/FixedPointMathLib.sol";
+    `;
+
+      const transformedContent = transformRemappings(fileContent);
+
+      expect(transformedContent).to.include(`import 'some-package/Contract.sol';`);
+      expect(transformedContent).to.include(`import 'another-package/Token.sol';`);
+      expect(transformedContent).to.include(
+        `import {FixedPointMathLib} from 'solmate/src/utils/FixedPointMathLib.sol';`,
+      );
+      expect(transformedContent).to.include(
+        `import {FixedPointMathLib} from "solmate/src/utils/FixedPointMathLib.sol";`,
+      );
+      expect(transformedContent).to.include(
+        `import {FixedPointMathLib as FPL} from "solmate/src/utils/FixedPointMathLib.sol";`,
+      );
+      expect(transformedContent).to.include(`import * as FPL from "solmate/src/utils/FixedPointMathLib.sol";`);
+    });
+  });
+
+  describe('with remapping for path inside node_modules', function () {
+    before(() => {
+      mock({
+        'remappings.txt': `
+          solmate/=node_modules/solmate/src/`,
+      });
+    });
+
+    it('should apply remapping and then remove node_modules from path', () => {
```

Comment:
Isn't this test a duplicate of `should remove node_modules from import paths`?

Reply:
not really since the preconditions are different, the testcase defined in line 48 is only concerned with removing `node_modules` from import paths, with the precondition of no remapping matching the paths with `node_modules` in it  (expressed as `with remapping for local file only`)

this testcase, however, ensures that _when_ a remapping matches a path inside `node_modules`, the remapping is applied  _before_ removing `node_modules` from the path

feedback of course welcome on ways to structure this differently so the above is clear from reading the code alone